### PR TITLE
remove unused dependencies and upgrade packages

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?
- use  commons-fileupload2 instead of  commons-fileupload
- Bump apache-commons-logging version
- Remove unused apache-commons-el dependency
- Remove unused jetty dependencies

changes tested in https://ci.suse.de/view/Manager/view/Manager-Test/job/manager-TEST-Hexagon-acceptance-tests/1551/ 
## GUI diff

No difference.

- [x] **DONE**

## Documentation
No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- https://ci.suse.de/view/Manager/view/Manager-Test/job/manager-TEST-Hexagon-acceptance-tests/1551/ 
- [x] **DONE**

## Links

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Comment by mbussolotto** [comment]
not sure if it's really unused or not, added reviewers who may know it

**Comment by mcalmer** [comment]
> not sure if it's really unused or not, added reviewers who may know it

@witekest hopefully knows this for sure.

**Review by mcalmer — APPROVED** [review]
Ok. Please remember to inform rel-eng that we maybe able to drop "apache-commons-el" from our product.

**Comment by mbussolotto** [comment]
@witekest do you know if we can remove 
```
artifact: simpleclient_servlet
package: prometheus-client-java
```
acceptance tests looks good, but I'm not sure if we're missing any monitoring corner case

**Review by cbosdo — DISMISSED** [review]

**Comment by witekest** [comment]
> @witekest do you know if we can remove
> 
> ```
> artifact: simpleclient_servlet
> package: prometheus-client-java
> ```
> 
> acceptance tests looks good, but I'm not sure if we're missing any monitoring corner case

We need it for instrumentation. The endpoint `/rhn/metrics` is not available without it.

image

**Comment by mbussolotto** [comment]
thanks. Added again. @uyuni-project/release-engineering is there any pitfall on taking `commons-logging` from Leap_sle instead of Leap?

File: java/buildconf/ivy/obs-maven-config.yaml [hunk 1]
```diff
@@ -79,7 +76,7 @@ artifacts:
   - artifact: commons-logging
     package: apache-commons-logging
     jar: apache-commons-logging\.jar
-    repository: Leap
+    repository: Leap_sle
   - artifact: commons-validator
     package: apache-commons-validator
```
Comment (deneb-alpha) [inline java/buildconf/ivy/obs-maven-config.yaml:84]: https://smelt.suse.de/maintained/?q=apache-commons-validator

I don't see it in the channels and I don't see it in the SLE codestreams either. you can make the switch if that source is somehow available in SLE. 

Am I missing something?

**Review by deneb-alpha — CHANGES_REQUESTED** [review]
Something doesn't look good I think

File: java/buildconf/ivy/obs-maven-config.yaml (hunk 1 above)
Comment (mbussolotto) [reply java/buildconf/ivy/obs-maven-config.yaml:84]: yes, the line. The package is `apache-commons-logging` not `apache-commons-validator`.  https://smelt.suse.de/maintained/?q=apache-commons-logging

**Comment by mbussolotto** [comment]
@deneb-alpha CYPR ?

**Comment by mbussolotto** [comment]
put it in draft since it's waiting for a review for too long, so I'm going to add other changes on it

**Review by cbosdo — COMMENTED** [review]
It looks good to me but cannot be merged until you had the dependencies updates in Other

File: java/buildconf/ivy/obs-maven-config.yaml (hunk 1 above)
Comment (deneb-alpha) [reply java/buildconf/ivy/obs-maven-config.yaml:84]: I'm blind , sorry

**Comment by deneb-alpha** [comment]
@mbussolotto , could you please fill the PR description? the documentation section is not filled

**Comment by mbussolotto** [comment]
> @mbussolotto , could you please fill the PR description? the documentation section is not filled

I cleaned it up, but I don't think documentation is needed
